notebooks/Batchelor.ipynb

```diff
@@ -0,0 +1,348 @@
+    "def u_exact_np(x):\n",
+    "    u = -1\n",
+    "    theta = np.arctan2(x[1],x[0])\n",
+    "\n",
+    "    d_psi_d_r = (u/((1/4)*(np.pi**2)-1))*((-1/4)*(np.pi**2)*(np.sin(theta))+((1/2)*(np.pi)*(theta)*(np.sin(theta)))+((theta)*(np.cos(theta))))\n",
+    "    d_psi_d_theta_overR = ((u)/((1/4)*(np.pi**2)-1)*(((-1/4)*(np.pi**2)*np.cos(theta))+((1/2)*(np.pi)*(np.sin(theta)))+((1/2)*(np.pi)*(theta)*(np.cos(theta)))+(np.cos(theta))-((theta)*(np.sin(theta)))))\n",
+    "\n",
+    "    return np.array([(((np.cos(theta) * (d_psi_d_theta_overR))) + (np.sin(theta) * d_psi_d_r)), (((np.sin(theta) * (d_psi_d_theta_overR))) - (np.cos(theta) * d_psi_d_r))])\n",
+    "\n",
+    "def u_exact(x):\n",
+    "    u = -1\n",
+    "    theta = ufl.atan2(x[1],x[0])\n",
+    "\n",
+    "    d_psi_d_r = (u/((1/4)*(ufl.pi**2)-1))*((-1/4)*(ufl.pi**2)*(ufl.sin(theta))+((1/2)*(ufl.pi)*(theta)*(ufl.sin(theta)))+((theta)*(ufl.cos(theta))))\n",
+    "    d_psi_d_theta_overR = ((u)/((1/4)*(ufl.pi**2)-1)*(((-1/4)*(ufl.pi**2)*ufl.cos(theta))+((1/2)*(ufl.pi)*(ufl.sin(theta)))+((1/2)*(ufl.pi)*(theta)*(ufl.cos(theta)))+(ufl.cos(theta))-((theta)*(ufl.sin(theta)))))\n",
+    "\n",
+    "    return ufl.as_vector([(((ufl.cos(theta) * (d_psi_d_theta_overR))) + (ufl.sin(theta) * d_psi_d_r)), (((ufl.sin(theta) * (d_psi_d_theta_overR))) - (ufl.cos(theta) * d_psi_d_r))])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4732c4cf-013f-4f2a-9071-85b408317528",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define a function which returns values for Dirichlet boundary condition (60)\n",
+    "def u_0(x):\n",
+    "    u = 1\n",
+    "    us = u * np.ones_like(x[0])\n",
+    "    zeroes = np.zeros_like(x[0])\n",
+    "    \n",
+    "    return np.array([us, zeroes])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2d81ad31-0597-42d4-ad0e-ac770045eedb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def batchelor(ne, p=1):\n",
+    "    # define the domain (unit square) of length and width ne, running in parallel\n",
+    "    domain = mesh.create_unit_square(MPI.COMM_WORLD, ne, ne)\n",
+    "\n",
+    "    # define velocity and pressure element\n",
+    "    v_e = basix.ufl.element(\"Lagrange\", domain.basix_cell(), p+1, shape=(domain.geometry.dim,))\n",
+    "    p_e = basix.ufl.element(\"Lagrange\", domain.basix_cell(), p)\n",
+    "\n",
+    "    # define mixed element\n",
+    "    vPe = basix.ufl.mixed_element([v_e, p_e])\n",
+    "\n",
+    "    # define function space\n",
+    "    V = functionspace(domain, vPe)\n",
+    "\n",
+    "    # define Velocity and Pressure function space\n",
+    "    V_v, _ = V.sub(0).collapse()\n",
+    "    V_p, _ = V.sub(1).collapse()\n",
+    "\n",
+    "    # define test functions\n",
+    "    v_t, p_t = ufl.TestFunctions(V)\n",
+    "\n",
+    "    # define trial functions\n",
+    "    trial_v, trial_p = ufl.TrialFunctions(V)\n",
```

Comment:
I know that `trial_v` and `trial_p` were helpful to you to keep track of what the variables were, but would it now be possible to switch them to `v_a` and `p_a` respectively for consistency with the paper notation?  Thanks!